Update transformer2d.md title

salcc opened the pull request:
For the other classes (e.g., UNet2DModel) the title of the documentation coincides with the name of the class, but that was not the case for Transformer2DModel.

stevhliu reviewed (commented):
Thanks! Can you also update it in the `toctree` as well? Let's also update TransformerTemporal and PriorTransformer to follow this convention.

HuggingFaceDocBuilderDev commented:
The docs for this PR live [here](https://moon-ci-docs.huggingface.co/docs/diffusers/pr_8375). All of your documentation changes will be reflected on that endpoint. The docs are available until 30 days after the last update.

salcc commented:
I updated TransformerTemporalModel and also a few more models that also had "Model" in the class name but not in the docs title.

PriorTransformer class name does not contain Model so I just removed the space but did not add the Model because that would require to change the code as well, which could break some code I guess.

stevhliu reviewed (approved):
Very nice, thank you for also updating some of the other models like PixArt and DiT!
